### ISSUE TITLE
Center in game window at the start of the program.

### ISSUE THREAD

**Issue by DasOhmoff** [ocornut/imgui#1694]
Hello, first of all I just want to mention that this is a great library and that I really like it :)
That being said, I have a little problem, I would like to know how to set the window position automatically to the middle of the screen on the start of the program.

My code:
```cpp
ImGui::SetNextWindowPosCenter(ImGuiCond_Once | ImGuiCond_Appearing);
ImGui::Begin("Test Window.", &showWindow, ImGuiWindowFlags_AlwaysAutoResize | ImGuiWindowFlags_NoCollapse | ImGuiWindowFlags_ShowBorders);

ImGui::InputText("World Width", width, 6);
ImGui::InputText("World Height", height, 6);
            
ImGui::Button("Cancel");
ImGui::SameLine();
ImGui::Button("Create");

ImGui::End();
```

The problem is that `ImGuiCond_Once` does not seem to work as I though it would. Here is my screen when I open the test window in the program for the first time: 
<img src="https://i.imgur.com/a6fd7Tt.png"/>

After I close the test window in the program and reopen it, it works just how it is supposed to work.
<img src="https://i.imgur.com/3b1r3gG.png"/>

So basically to center the test window, I have to close and reopen it.
How do I set the test window position to the center without having to close and reopen it?

**Comment by lachlanpage** [comment]
from #1122 

"You can position windows using SetNextWindowPos(). You can disable window moving, and within a window use SetCursorScreenPos() to position the next widgets. Please read on imgui_demo.cpp, execute the demo app and read imgui.h to find details." 

or you can manually edit the imgui.ini file

**Comment by DasOhmoff** [comment]
This is not the problem...

**Comment by ocornut** [comment]
Hello @DasOhmoff, thanks for reporting this. Looks like something broke since I introduced this feature (in this situation the positioning is applied before the window had a chance to calculate its own size).

Will look into it..

(PS 1: you are using an old version, `SetNextWindowPosCenter()` was obsolete in September)
(PS 2: combining multiple Cond as flags is described as illegal `don't combine multiple values into flags`. this is not the cause of your issue but letting you know I will change this to the ImGuiCond values become linear values, as described in the comments)

**Comment by DasOhmoff** [comment]
Thank you my friend :) .

**Comment by ocornut** [comment]
This is an unusually complex/tricky matter, so although I have no pushed a fix which I believe will fix your issue, it probably has subtle side effects on other uses and I expect to return to it at some point. For now I believe your issue is fixed if you grab latest.

FYI I believe `SetNextWindowPosCenter()` on appearing never worked with non-popup/modal windows in the first place, and this is how I tested it (with a modal).
